sdk/storage/azure-storage-blob/azure/storage/blob/aio/_download_async.py

```diff
@@ -231,26 +237,53 @@ def __init__(
         self._file_size = None
         self._non_empty_ranges = None
         self._response = None
+        self._encryption_data = None
+
+        self._initial_range = None
+        self._initial_offset = None
 
         # The service only provides transactional MD5s for chunks under 4MB.
         # If validate_content is on, get only self.MAX_CHUNK_GET_SIZE for the first
         # chunk so a transactional MD5 can be retrieved.
         self._first_get_size = self._config.max_single_get_size if not self._validate_content \
             else self._config.max_chunk_get_size
+
+    def __len__(self):
+        return self.size
+
+    async def _get_encryption_data_request(self):
+        # Save current request cls
+        download_cls = self._request_options.pop('cls', None)
+        # Adjust cls for get_properties
+        self._request_options['cls'] = deserialize_blob_properties
+
+        properties = await self._clients.blob.get_properties(**self._request_options)
+        # This will return None if there is no encryption metadata or there are parsing errors.
+        # That is acceptable here, the proper error will be caught and surfaced hen attempting
+        # to decrypt the blob.
+        self._encryption_data = parse_encryption_data(properties.metadata)
+
+        # Restore cls for download
+        self._request_options['cls'] = download_cls
+
+    async def _setup(self):
+        if self._encryption_options.get("key") is not None or self._encryption_options.get("resolver") is not None:
```

Comment:
Do we also need to check `requires_encryption=True`? Or is that assessed elsewhere?

Reply:
`requires_encryption` is sort of an odd option. In almost all cases its mostly ignored and whether we encrypt/decrypt is controlled by whether the user passes `key_encryption_key` or `key_resolver_function` on client construction. Even if you set this to `False` but provide a `key_encryption_key`, we would still perform encryption.

The only case this value is taken into account is if the blob isn't actually encrypted (no encryption metadata is found) or there is an error in parsing the metadata. If `requires_encryption` is `True` these cases will cause an exception, if its `False`, no exception will occur, and the blob will be treated as unencrypted.

So to actually answer your question, no I don't think we need/want to assess that value here since if someone sets this to `True` for some reason but doesn't provide the other arguments, we don't want to make this extra request.